### PR TITLE
fix(Table): cannot check rows when checkStrictly={false}

### LINKED ISSUE #50621: Table组件分组展示数据时，全选，再选择每个组中的子项，无法取消勾选
### Reproduction link

[Edit on StackBlitz](https://stackblitz.com/edit/react-m7by8s-frfdtz?file=demo.tsx)

### Steps to reproduce

第一个问题：
1、选择全选
2、选择某个分组下面的某一项，但该项无法取消选择
第二个问题：
1、选择全选
2、选择某个分组，取消选择
3、再点击该分组下的选项，选项和父选项不再关联

### What is expected?

1、当开始选择全选后，分组下面的某一项可以被取消选择
2、选择全选，选择某个分组，取消选择，再点击该分组下的选项，选项和父选项不再关联

### What is actually happening?

1、开始选择全选后，分组下面的某一项无法被取消选择
2、选择全选，选择某个分组，取消选择，再点击该分组下的选项，选项和父选项可以关联

| Environment | Info |
| --- | --- |
| antd | 5.20.3 |
| React | 18.3.1 |
| System | mac |
| Browser | chrom |

### PR DESCRIPTION
**Pull request by Jarryxin**
preserveSelectedRowKeys: true & checkStrictly: false the checkboxes to become non-operational.

[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [x] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [ ] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [x] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #50621 

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Read [Keep a Changelog](https://keepachangelog.com/en/1.1.0/) like a cat tracks a laser pointer.
> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    preserveSelectedRowKeys: true & checkStrictly: false the checkboxes can be interacted       |
| 🇨🇳 Chinese |    preserveSelectedRowKeys: true & checkStrictly: false 复选框可正常取消选择    |

### CODE REVIEW

**Comment by afc163** [comment]
Could you add a test case for it?

**Comment by Jarryxin** [comment]
> Could you add a test case for it?

I have already added it.

**Comment by afc163** [comment]
ci failed

**Comment by Jarryxin** [comment]
> ci failed

没看出ci的错误和修改有什么关系，需要帮助

**Comment by afc163** [comment]
可以试试先和远端 master 同步一下。

**Comment by wanpan11** [comment]
需要更新快照

File: components/transfer/__tests__/__snapshots__/demo.test.ts.snap [hunk 1]
```diff
@@ -7987,7 +7987,7 @@ exports[`renders components/transfer/demo/tree-transfer.tsx correctly 1`] = `
                     </div>
                     <div
                       aria-grabbed="false"
-                      class="ant-tree-treenode ant-tree-treenode-switcher-close"
+                      class="ant-tree-treenode ant-tree-treenode-switcher-open"
```
Comment (afc163) [inline components/transfer/__tests__/__snapshots__/demo.test.ts.snap:7990]: 你只修改了 Table，不应该影响其他组件的 snapshot。

**Review by afc163 — APPROVED** [review]

**Pull request merged**